// @azure/monitor-opentelemetry-exporter/dist/commonjs/types.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaxPropertyLengths = exports.BreezePerformanceCounterNames = exports.OTelPerformanceCounterNames = void 0;
/**
 * Performance Counter OpenTelemetry compliant names.
 * @internal
 */
var OTelPerformanceCounterNames;
(function (OTelPerformanceCounterNames) {
    OTelPerformanceCounterNames["PRIVATE_BYTES"] = "Private_Bytes";
    OTelPerformanceCounterNames["AVAILABLE_BYTES"] = "Available_Bytes";
    OTelPerformanceCounterNames["PROCESSOR_TIME"] = "Processor_Time";
    OTelPerformanceCounterNames["PROCESS_TIME"] = "Process_Time";
    OTelPerformanceCounterNames["REQUEST_RATE"] = "Request_Rate";
    OTelPerformanceCounterNames["REQUEST_DURATION"] = "Request_Execution_Time";
})(OTelPerformanceCounterNames || (exports.OTelPerformanceCounterNames = OTelPerformanceCounterNames = {}));
/**
 * Breeze Performance Counter names.
 * @internal
 */
var BreezePerformanceCounterNames;
(function (BreezePerformanceCounterNames) {
    BreezePerformanceCounterNames["PRIVATE_BYTES"] = "\\Process(??APP_WIN32_PROC??)\\Private Bytes";
    BreezePerformanceCounterNames["AVAILABLE_BYTES"] = "\\Memory\\Available Bytes";
    BreezePerformanceCounterNames["PROCESSOR_TIME"] = "\\Processor(_Total)\\% Processor Time";
    BreezePerformanceCounterNames["PROCESS_TIME"] = "\\Process(??APP_WIN32_PROC??)\\% Processor Time";
    BreezePerformanceCounterNames["REQUEST_RATE"] = "\\ASP.NET Applications(??APP_W3SVC_PROC??)\\Requests/Sec";
    BreezePerformanceCounterNames["REQUEST_DURATION"] = "\\ASP.NET Applications(??APP_W3SVC_PROC??)\\Request Execution Time";
})(BreezePerformanceCounterNames || (exports.BreezePerformanceCounterNames = BreezePerformanceCounterNames = {}));
/**
 * Property Max Lengths
 * @internal
 */
var MaxPropertyLengths;
(function (MaxPropertyLengths) {
    MaxPropertyLengths[MaxPropertyLengths["NINE_BIT"] = 512] = "NINE_BIT";
    MaxPropertyLengths[MaxPropertyLengths["TEN_BIT"] = 1024] = "TEN_BIT";
    MaxPropertyLengths[MaxPropertyLengths["THIRTEEN_BIT"] = 8192] = "THIRTEEN_BIT";
    MaxPropertyLengths[MaxPropertyLengths["FIFTEEN_BIT"] = 32768] = "FIFTEEN_BIT";
})(MaxPropertyLengths || (exports.MaxPropertyLengths = MaxPropertyLengths = {}));
